Automattic/woocommerce-payments
# Query the Server to fetch new rates when the store default currency is changed

Fixes #2073 

#### Changes proposed in this Pull Request









- When the store default currency is updated via the WooCommerce settings, update the default Multi Currency currency, fetch new rates from the server based on the new default currency and reinitialise the available/enabled currencies.

Some considerations: 
- Is this the best hook to use to identify the store currency changing? 
- Are there any other ways the default store currency could be changed that we should check for?

#### Testing instructions





* Update your sandbox to the latest server code by running `svn up .` from the `~/public_html` folder within your sandbox.
* Point your client to your sandbox by disabling API redirection in the WCPay Dev Tools.
* Go to `Settings -> Multi-Currency` and add a few currencies (e.g. EUR, GBP, CAD, VND) to the enabled currencies list using the `Add Currencies` modal. (If you have already got a few enabled currencies, skip this step).
* Go to `Settings -> General` and change the default currency of the store from USD to EUR.
* Go to `Settings -> Multi-Currency`. Check that the default currency is Euros, and the rates displayed for the other currencies are roughly in line with the current currency conversion rates between Euros and each currency.
* Go to `Settings -> General` and change the default currency of the store back to USD.
* Go back to the `Settings -> Multi-Currency` page and verify that USD is now the default currency, and the rates being displayed are now in line with the conversion rates between USD and each currency.

Notes: I opened a [follow-up issue](https://github.com/Automattic/woocommerce-payments/issues/2316) for some issues I noticed related to zero-decimal currencies which I found while working on this issue. 

-------------------
